File: public/docs/_examples/toh-5/dart/lib/hero_detail_component.dart
// #docplaster
// #docregion , v2
// #docregion added-imports
import 'dart:async';

// #enddocregion added-imports
import 'package:angular2/core.dart';
// #docregion added-imports
import 'package:angular2/router.dart';
import 'package:angular2/platform/common.dart';

// #enddocregion added-imports
import 'hero.dart';
// #docregion added-imports
import 'hero_service.dart';
// #enddocregion added-imports

@Component(
    selector: 'my-hero-detail',
    // #docregion metadata, templateUrl
    templateUrl: 'hero_detail_component.html',
    // #enddocregion metadata, templateUrl, v2
    styleUrls: const ['hero_detail_component.css']
    // #docregion v2
    )
// #docregion implement
class HeroDetailComponent implements OnInit {
  // #enddocregion implement
  Hero hero;
  // #docregion ctor
  final HeroService _heroService;
  final RouteParams _routeParams;
  final Location _location;

  HeroDetailComponent(this._heroService, this._routeParams, this._location);
  // #enddocregion ctor

  // #docregion ngOnInit
  Future<Null> ngOnInit() async {
    var _id = _routeParams.get('id');
    var id = int.parse(_id ?? '', onError: (_) => null);
    if (id != null) hero = await (_heroService.getHero(id));
  }
  // #enddocregion ngOnInit

  // #docregion goBack
  void goBack() => _location.back();
  // #enddocregion goBack
}

File: public/docs/_examples/toh-5/ts/app/hero-detail.component.ts
// #docplaster
// #docregion
// #docregion import-oninit, v2
import { Component, OnInit, OnDestroy } from '@angular/core';
// #enddocregion import-oninit
// #docregion import-activated-route
import { ActivatedRoute } from '@angular/router';
// #enddocregion import-activated-route

import { Hero } from './hero';
// #docregion import-hero-service
import { HeroService } from './hero.service';
// #enddocregion import-hero-service

// #docregion extract-template
@Component({
  selector: 'my-hero-detail',
  // #docregion template-url
  templateUrl: 'app/hero-detail.component.html',
  // #enddocregion template-url, v2
  styleUrls: ['app/hero-detail.component.css']
  // #docregion v2
})
// #enddocregion extract-template
// #docregion implement
export class HeroDetailComponent implements OnInit, OnDestroy {
  // #enddocregion implement
  hero: Hero;
  sub: any;

  // #docregion ctor
  constructor(
    private heroService: HeroService,
    private route: ActivatedRoute) {
  }
  // #enddocregion ctor

  // #docregion ng-oninit
  ngOnInit() {
    // #docregion get-id
    this.sub = this.route.params.subscribe(params => {
      let id = +params['id'];
      this.heroService.getHero(id)
        .then(hero => this.hero = hero);
    });
    // #enddocregion get-id
  }
  // #enddocregion ng-oninit

  // #docregion ng-ondestroy
  ngOnDestroy() {
    this.sub.unsubscribe();
  }
  // #enddocregion ng-ondestroy

  // #docregion go-back
  goBack() {
    window.history.back();
  }
// #enddocregion go-back
}
